lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -122,6 +157,16 @@ findModule(ASTContext &ctx, AccessPathElem moduleID,
                             archFile.str(), archDocFile.str(),
                             moduleBuffer, moduleDocBuffer,
                             scratch);
+
+    if(err == std::errc::no_such_file_or_directory) {
+      addDiagnosticInfoForArchitectureMismatch(ctx,
+                                               moduleID.second,
+                                               moduleName,
+                                               archName,
+                                               currPath);
+    }
+
+    return false;
```

Comment:
I think this was supposed to go inside your new `if` statement, right?

Reply:
actually I don't need the `return false;` statement. 

The following line should suffice
```cpp
if (!err)
  return true;
```